cmd/manager/main.go

```diff
@@ -135,11 +136,23 @@ func main() {
 	if strings.ToLower(os.Getenv("ENABLE_WEBHOOKS")) == "true" {
 		log.Printf("Setting up webhooks")
 
-		if err := registerWebhook("/admit", &v1beta1.Instance{}, &webhook.Admission{Handler: &kudohook.InstanceAdmission{}}, mgr); err != nil {
+		// client.New returns a new Client using the provided config and Options.
+		// The returned client reads *and* writes directly from the server
+		// (it doesn't use object caches). Using a cached client might lead to racy
+		// behaviour when installing operators e.g. and `OperatorVersion` is already created
+		// but not yet in cache which leads to an error during `Instance` creation.
+		uncachedClient, err := client.New(mgr.GetConfig(), client.Options{Scheme: mgr.GetScheme()})
+		if err != nil {
+			log.Printf("Unable to create an uncached client for the webhook: %v", err)
+			os.Exit(1)
+		}
+
+		iac := &webhook.Admission{Handler: &kudohook.InstanceAdmission{Client: uncachedClient}}
+		if err := registerWebhook("/admit", &v1beta1.Instance{}, iac, mgr); err != nil {
```

Comment:
I was questioning why `InstanceAdmission.Client` was made public... it comes down to the code above and the need to "set" the client.     Additionally the `uncachedClient` is only used for IA... and specifically the godocs above the creation of `uncachedClient` is really specific to the IA (and seems out of place in Main).   

Could we invert this?  which would clean up all this...

if we have `webhook.New` or `webhook.NewAdmission`  passing config and scheme then all this ends up in the IA code and we do not need to break the encapsulation on `Client`.

resulting in something close to the original code like: 

```
if err := registerWebhook("/admit", &v1beta1.Instance{}, &webhook.Admission{Handler: &kudohook.NewAdmission(mgr.GetConfig(), mgr.GetScheme())}, mgr); err != nil {
```

of course it still might make more sense to return (*InstanceAdmission, error) with an error resulting in 2 separate lines:

```
		iac, err := webhook.NewAdmission(mgr.GetConfig(), mgr.GetScheme())
		if err != nil {
			log.Printf("Unable to create instance admission webhook: %v", err)
			os.Exit(1)
		}
		if err := registerWebhook("/admit", &v1beta1.Instance{}, iac, mgr); err != nil {
			log.Printf("Unable to register instance admission webhook: %v", err)
			os.Exit(1)
		}

```

Reply:
Encapsulation ftw! ❤️